# **Pandas Library Practice in Python**

**Import Library**

In [ ]:
import pandas as pd 

**Import Data**

In [ ]:
#Read Data by Jupyter
#df = pd.read_csv("./HNP_StatsData.csv")
#df.head()

In [ ]:
#Read Data Zip by Colab
!unzip /content/HNP_StatsData.zip -d /content/HNP_StatsData

Archive:  /content/HNP_StatsData.zip
  inflating: /content/HNP_StatsData/HNP_StatsData.csv  
  inflating: /content/HNP_StatsData/__MACOSX/._HNP_StatsData.csv  


In [ ]:
df = pd.read_csv('/content/HNP_StatsData/HNP_StatsData.csv')
df.head()

Country Name Country Code  \
0  Africa Eastern and Southern          AFE   
1  Africa Eastern and Southern          AFE   
2  Africa Eastern and Southern          AFE   
3  Africa Eastern and Southern          AFE   
4  Africa Eastern and Southern          AFE   

                                      Indicator Name  Indicator Code  \
0  Adolescent fertility rate (births per 1,000 wo...     SP.ADO.TFRT   
1  Adults (ages 15+) and children (0-14 years) li...     SH.HIV.TOTL   
2  Adults (ages 15+) and children (ages 0-14) new...  SH.HIV.INCD.TL   
3                  Adults (ages 15+) living with HIV     SH.DYN.AIDS   
4        Adults (ages 15-49) newly infected with HIV     SH.HIV.INCD   

         1960        1961        1962       1963       1964        1965  ...  \
0  140.120535  140.594564  141.128109  141.32344  141.50816  141.668896  ...   
1         NaN         NaN         NaN        NaN        NaN         NaN  ...   
2         NaN         NaN         NaN        NaN        NaN         NaN  ...   
3         NaN         NaN         NaN        NaN        NaN         NaN  ...   
4         NaN         NaN         NaN        NaN        NaN         NaN  ...   

         2013        2014       2015       2016       2017       2018  \
0  103.391461  101.541782  99.719799  97.916601  96.146492  94.695223   
1         NaN         NaN        NaN        NaN        NaN        NaN   
2         NaN         NaN        NaN        NaN        NaN        NaN   
3         NaN         NaN        NaN        NaN        NaN        NaN   
4         NaN         NaN        NaN        NaN        NaN        NaN   

        2019  2020  2021  Unnamed: 66  
0  93.231944   NaN   NaN          NaN  
1        NaN   NaN   NaN          NaN  
2        NaN   NaN   NaN          NaN  
3        NaN   NaN   NaN          NaN  
4        NaN   NaN   NaN          NaN  

[5 rows x 67 columns]

**Q1: Which country has the highest rate of mortality by road traffic injury in 2019?**

In [ ]:
df_indicator_name = df["Indicator Name"]
list_indicators = list(df_indicator_name)

In [ ]:
def search_spesific_indicator(indicator_like, indicator_list):
    tmp = []
    for indicator in indicator_list:
        if indicator_like.lower() in indicator.lower():
            tmp.append(indicator)
    return set(tmp) 
indicator = list(search_spesific_indicator("traffic", list_indicators))
print(indicator)

['Mortality caused by road traffic injury, female (per 100,000 female population)', 'Mortality caused by road traffic injury (per 100,000 people)', 'Mortality caused by road traffic injury, male (per 100,000 male population)']


In [ ]:
tmp = df[(df['Indicator Name'].isin(indicator))]
df_target_mortality = tmp[['Country Name', 'Country Code', 'Indicator Name', '2019']]
target = ['Mortality caused by road traffic injury (per 100,000 people)']
df_target_mortality = df_target_mortality[df_target_mortality["Indicator Name"].isin(target)]
df_target_mortality

Country Name Country Code  \
163        Africa Eastern and Southern          AFE   
604         Africa Western and Central          AFW   
1045                        Arab World          ARB   
1486            Caribbean small states          CSS   
1927    Central Europe and the Baltics          CEB   
...                                ...          ...   
115264           Virgin Islands (U.S.)          VIR   
115705              West Bank and Gaza          PSE   
116146                     Yemen, Rep.          YEM   
116587                          Zambia          ZMB   
117028                        Zimbabwe          ZWE   

                                           Indicator Name       2019  
163     Mortality caused by road traffic injury (per 1...  29.380510  
604     Mortality caused by road traffic injury (per 1...  24.568777  
1045    Mortality caused by road traffic injury (per 1...  20.170676  
1486    Mortality caused by road traffic injury (per 1...  14.454147  
1927    Mortality caused by road traffic injury (per 1...   8.583368  
...                                                   ...        ...  
115264  Mortality caused by road traffic injury (per 1...        NaN  
115705  Mortality caused by road traffic injury (per 1...        NaN  
116146  Mortality caused by road traffic injury (per 1...  29.400000  
116587  Mortality caused by road traffic injury (per 1...  20.500000  
117028  Mortality caused by road traffic injury (per 1...  41.200000  

[266 rows x 4 columns]

In [ ]:
answer_1 = df_target_mortality.loc[df_target_mortality['2019'].idxmax()]
answer_1

Country Name                                     Dominican Republic
Country Code                                                    DOM
Indicator Name    Mortality caused by road traffic injury (per 1...
2019                                                           64.6
Name: 46468, dtype: object

**Dominican Republic** has the highest rate of mortality by road traffic injury in 2019.



**Q2: Which country has 67% of their pregnant women receiving prenatal care in 2018?**


In [ ]:
indicator = list(search_spesific_indicator("prenatal", list_indicators))
print(indicator)

['Pregnant women receiving prenatal care (%)', 'Pregnant women receiving prenatal care of at least four visits (% of pregnant women)', 'Cause of death, by communicable diseases and maternal, prenatal and nutrition conditions (% of total)']


In [ ]:
tmp = df[(df['Indicator Name'].isin(indicator))]
df_target_prenatal = tmp[['Country Name', 'Country Code', 'Indicator Name', '2018']]
df_target_prenatal

Country Name Country Code  \
69      Africa Eastern and Southern          AFE   
330     Africa Eastern and Southern          AFE   
331     Africa Eastern and Southern          AFE   
510      Africa Western and Central          AFW   
771      Africa Western and Central          AFW   
...                             ...          ...   
116754                       Zambia          ZMB   
116755                       Zambia          ZMB   
116934                     Zimbabwe          ZWE   
117195                     Zimbabwe          ZWE   
117196                     Zimbabwe          ZWE   

                                           Indicator Name  2018  
69      Cause of death, by communicable diseases and m...   NaN  
330            Pregnant women receiving prenatal care (%)   NaN  
331     Pregnant women receiving prenatal care of at l...   NaN  
510     Cause of death, by communicable diseases and m...   NaN  
771            Pregnant women receiving prenatal care (%)   NaN  
...                                                   ...   ...  
116754         Pregnant women receiving prenatal care (%)  96.9  
116755  Pregnant women receiving prenatal care of at l...  63.5  
116934  Cause of death, by communicable diseases and m...   NaN  
117195         Pregnant women receiving prenatal care (%)   NaN  
117196  Pregnant women receiving prenatal care of at l...   NaN  

[798 rows x 4 columns]

In [ ]:
target = ['Pregnant women receiving prenatal care (%)']
df_target_prenatal = df_target_prenatal[df_target_prenatal["Indicator Name"].isin(target)]
df_target_prenatal.head()

Country Name Country Code  \
330      Africa Eastern and Southern          AFE   
771       Africa Western and Central          AFW   
1212                      Arab World          ARB   
1653          Caribbean small states          CSS   
2094  Central Europe and the Baltics          CEB   

                                  Indicator Name  2018  
330   Pregnant women receiving prenatal care (%)   NaN  
771   Pregnant women receiving prenatal care (%)   NaN  
1212  Pregnant women receiving prenatal care (%)   NaN  
1653  Pregnant women receiving prenatal care (%)   NaN  
2094  Pregnant women receiving prenatal care (%)   NaN

In [ ]:
answer_2 = df_target_prenatal[df_target_prenatal["2018"] == 67.0]
answer_2

Country Name Country Code                              Indicator Name  \
85443      Nigeria          NGA  Pregnant women receiving prenatal care (%)   

       2018  
85443  67.0

**Nigeria** has 67% of their pregnant women receiving prenatal care in 2018.

**Q3: How much is the prevalence of overweight adults in the United States, in 2016?**

In [ ]:
indicator = list(search_spesific_indicator("overweight", list_indicators))
print(indicator)

['Prevalence of overweight, female (% of children under 5)', 'Prevalence of overweight (% of children under 5)', 'Prevalence of overweight, male (% of children under 5)', 'Prevalence of overweight, female (% of female adults)', 'Prevalence of overweight (% of adults)', 'Prevalence of overweight, male (% of male adults)', 'Prevalence of overweight (modeled estimate, % of children under 5)']


In [ ]:
spesific_indicator = ['Prevalence of overweight (% of adults)']

In [ ]:
tmp = df[(df['Indicator Name'].isin(spesific_indicator))]
df_target_overweight = tmp[['Country Name', 'Country Code', 'Indicator Name', '2016']]
df_target_overweight

Country Name Country Code  \
345        Africa Eastern and Southern          AFE   
786         Africa Western and Central          AFW   
1227                        Arab World          ARB   
1668            Caribbean small states          CSS   
2109    Central Europe and the Baltics          CEB   
...                                ...          ...   
115446           Virgin Islands (U.S.)          VIR   
115887              West Bank and Gaza          PSE   
116328                     Yemen, Rep.          YEM   
116769                          Zambia          ZMB   
117210                        Zimbabwe          ZWE   

                                Indicator Name       2016  
345     Prevalence of overweight (% of adults)  28.917314  
786     Prevalence of overweight (% of adults)  28.868763  
1227    Prevalence of overweight (% of adults)  61.843016  
1668    Prevalence of overweight (% of adults)  53.123588  
2109    Prevalence of overweight (% of adults)  59.044895  
...                                        ...        ...  
115446  Prevalence of overweight (% of adults)        NaN  
115887  Prevalence of overweight (% of adults)        NaN  
116328  Prevalence of overweight (% of adults)  48.800000  
116769  Prevalence of overweight (% of adults)  27.800000  
117210  Prevalence of overweight (% of adults)  38.200000  

[266 rows x 4 columns]

In [ ]:
answer_3 = df_target_overweight[(df_target_overweight["Country Name"] == "United States")]
answer_3

Country Name Country Code                          Indicator Name  \
112800  United States          USA  Prevalence of overweight (% of adults)   

        2016  
112800  67.9

The prevalence of overweight adults of United States in 2016 is **67.9**

**Q4: Which country is more likely to have TBC due to the lowest rate of immunization in 2019?**

In [ ]:
indicator = list(search_spesific_indicator("immunization", list_indicators))
print(indicator)

['Immunization, BCG (% of one-year-old children)', 'Immunization, measles (% of children ages 12-23 months)', 'Immunization, measles second dose (% of children by the nationally recommended age)', 'Immunization, Hib3 (% of children ages 12-23 months)', 'Immunization, Pol3 (% of one-year-old children)', 'Immunization, HepB3 (% of one-year-old children)', 'Immunization, DPT (% of children ages 12-23 months)']


In [ ]:
spesific_indicator_4 = ['Immunization, BCG (% of one-year-old children)']

In [ ]:
tmp = df[(df['Indicator Name'].isin(spesific_indicator_4))]
df_target_imun = tmp[['Country Name', 'Country Code', 'Indicator Name', '2019']]
df_target_imun

Country Name Country Code  \
131        Africa Eastern and Southern          AFE   
572         Africa Western and Central          AFW   
1013                        Arab World          ARB   
1454            Caribbean small states          CSS   
1895    Central Europe and the Baltics          CEB   
...                                ...          ...   
115232           Virgin Islands (U.S.)          VIR   
115673              West Bank and Gaza          PSE   
116114                     Yemen, Rep.          YEM   
116555                          Zambia          ZMB   
116996                        Zimbabwe          ZWE   

                                        Indicator Name       2019  
131     Immunization, BCG (% of one-year-old children)  80.685117  
572     Immunization, BCG (% of one-year-old children)  77.126464  
1013    Immunization, BCG (% of one-year-old children)  86.995799  
1454    Immunization, BCG (% of one-year-old children)        NaN  
1895    Immunization, BCG (% of one-year-old children)  88.466252  
...                                                ...        ...  
115232  Immunization, BCG (% of one-year-old children)        NaN  
115673  Immunization, BCG (% of one-year-old children)        NaN  
116114  Immunization, BCG (% of one-year-old children)  69.000000  
116555  Immunization, BCG (% of one-year-old children)  95.000000  
116996  Immunization, BCG (% of one-year-old children)  95.000000  

[266 rows x 4 columns]

In [ ]:
answer_4 = df_target_imun.loc[df_target_imun['2019'].idxmin()]
answer_4

Country Name                                              Sweden
Country Code                                                 SWE
Indicator Name    Immunization, BCG (% of one-year-old children)
2019                                                        25.0
Name: 104207, dtype: object

**Sweden** is more likely to have TBC due to the lowest rate of immunization in 2019

**Q5: Which country has the average age of female at first marriage lower than 20 years old in 2016?**

In [ ]:
indicator = list(search_spesific_indicator("marriage", list_indicators))
print(indicator)

['Age at first marriage, male', 'Age at first marriage, female']


In [ ]:
spesific_indicator_5 = ['Age at first marriage, female']

In [ ]:
tmp = df[(df['Indicator Name'].isin(spesific_indicator_5))]
df_target_marriage = tmp[['Country Name', 'Country Code', 'Indicator Name', '2016']]
df_target_marriage

Country Name Country Code  \
5          Africa Eastern and Southern          AFE   
446         Africa Western and Central          AFW   
887                         Arab World          ARB   
1328            Caribbean small states          CSS   
1769    Central Europe and the Baltics          CEB   
...                                ...          ...   
115106           Virgin Islands (U.S.)          VIR   
115547              West Bank and Gaza          PSE   
115988                     Yemen, Rep.          YEM   
116429                          Zambia          ZMB   
116870                        Zimbabwe          ZWE   

                       Indicator Name  2016  
5       Age at first marriage, female   NaN  
446     Age at first marriage, female   NaN  
887     Age at first marriage, female   NaN  
1328    Age at first marriage, female   NaN  
1769    Age at first marriage, female   NaN  
...                               ...   ...  
115106  Age at first marriage, female   NaN  
115547  Age at first marriage, female   NaN  
115988  Age at first marriage, female   NaN  
116429  Age at first marriage, female   NaN  
116870  Age at first marriage, female   NaN  

[266 rows x 4 columns]

In [ ]:
answer_5 = df_target_marriage[(df_target_marriage["2016"] < 20)]
answer_5

Country Name Country Code                 Indicator Name  2016
74093       Malawi          MWI  Age at first marriage, female  19.9

**Malawi** has the average age of female at first marriage lower than 20 years old in 2016